include/seahorn/Expr/TypeChecker.hh

```diff
@@ -1,17 +1,168 @@
 #pragma once
-
-#include "seahorn/Expr/ExprApi.hh"
+#include "seahorn/Expr/ExprCore.hh"
+#include "seahorn/Expr/ExprOpSort.hh"
 
 namespace expr {
 class TypeChecker {
   class Impl;
-  Impl* m_impl;
+  Impl *m_impl;
 
 public:
   TypeChecker();
   ~TypeChecker();
   Expr sortOf(Expr e) { return this->typeOf(e); }
-  Expr typeOf(Expr e); 
+  Expr typeOf(Expr e);
+  Expr getErrorExp();         // to be called after typeOf() or sortOf()
+  ExprSet getEndExps(Expr e); // to be called after typeOf() or sortOf()
 };
 
+namespace op {
```

Comment:
These don't belong to the same header file. The code up to hear is likley to be included by every client of type checker. The code after here is only needed to implement type checking functionality. Maybe move it into `TypeCheckerDefs.hh` or `TypeCheckerUtils.hh`